### ISSUE TITLE
Tray Icon Missing on 5.4.4-beta.3

### ISSUE THREAD

**Issue by LucesEnBabel** [getferdi/ferdi#555]
**Describe the bug**

There's no tray icon in this version of Ferdi, Stable version does have it.

**To Reproduce**

1. Download the latest pre-release version.
2. Enable close to tray.

**Expected behavior**
It should show the application tray icon.

**Screenshots**
N/A

**Desktop:**

- OS: elementary OS 5.1.3 Hera x86_64
- Ferdi Version 5.4.4-beta.3
- Server Used: N/A

Debug information: https://debug.getferdi.com/dbefd5de-5d79-4ac9-82e8-b62483b6f1b1

**Additional context**

Probably related to Electron. (is a rollback possible?)

**Comment by vantezzen** [comment]
Related to #22550: ElectronJS tray icons seem to have problems with Elementary OS - can you try the workarounds suggested at https://github.com/electron/electron/issues/22550#issuecomment-596046596 ?

**Comment by g0r3** [comment]
I am having the same problem on Arch Linux/KDE.

**Comment by ElhadaryHisham** [comment]
same on MX Linux

**Comment by hosler** [comment]
I'm seeing the same on manjaro

**Comment by don-andruha** [comment]
I have the same problem on **Linux Mint 19.3 Cinnamon**

**Comment by g0r3** [comment]
Fixed for me with 5.5.0

**Comment by hosler** [comment]
issue persists on 5.5.0 for me on gentoo and manjaro using i3wm

**Comment by kris7t** [comment]
Interestingly, for me, after building the ea5a5932 from the `develop` branch with Electron 8.2.5, the tray icon appears on Waybar (I use sway as a Wayland compositor), if I run Ferdi in release mode (`ELECTRON_IS_DEV=0`). In dev mode, however, I do errors with calling `setContextMenu` on `null`. I also get similar errors when I toggle "Show Ferdi in system tray" to off.

When I restart Waybar (`org.kde.StatusNotifierWatcher` disappears from the session D-Bus, the reappears as the new Waybar instance is initialized), the tray icon disappears. However, it comes back when I toggle "Show Ferdi in system tray" to off and then back on. I guess we would have similar behavior in GNOME Shell whenever the screen is locked.

**Comment by kytwb** [comment]
@kris7t's patch has been made available in [v5.5.1-nightly.15](https://github.com/getferdi/nightlies/releases/tag/v5.5.1-nightly.15). Feedback welcome! 🙏

**Comment by hosler** [comment]
I still dont have tray icon using this nightly

**Comment by pcklink** [comment]
I'm on Linux Mint and after updating to 5.5.0 I lost my tray icon for Ferdi. Going back to 5.4.3 gives me my tray icon back but services refuse to load (tweetdeck, mattermost, hangouts). Nightly 5.5.1-nightly.15 has no tray ion either even though it was presented as a fix for this. Any further suggestions? I'm using Ferdi to keep track of multiple platforms without clutter. Not being able to keep the window in the systray is kind of a deal breaker.

**Comment by kris7t** [comment]
Ferdi 5.5.0 is build on an Electron version that has removed XEmbed tray icon support, so the tray icon can only be shown in desktop environments supporting the StatusNotifier protocol. Out of the box, the version Cinnamon in Mint does not support StatusNotifier (but afaik [the next release coming in Mint 20 will](https://winaero.com/blog/cinnamon-4-6-released-check-out-whats-new/)).

**Comment by pcklink** [comment]
Thanks. Let's hope Mint 20 comes soon and resolves things.

**Comment by pcklink** [comment]
It's actually surprisingly easily fixed (in Mint 19):
"System Settings" -> "General" -> "Enable support for indicators"

**Comment by kris7t** [comment]
Indeed, although now -- after having updated to a newer Electron version -- I am having the opposite problem, i.e. the tray icon is sometimes duplicated. Will try to look into that when I'll have a bit more time.

**Comment by fermmm** [comment]
I have Linux Mint 20 with Cinnamon 4.6.6 and Ferdi 5.5.1-nightly.15 and still no tray icon.
